packages/permissionless/errors/account.ts

```diff
@@ -0,0 +1,306 @@
+                "• If you are looking to use a paymaster to cover the gas fees, verify that the paymasterAndData field is set.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountSignatureExpiredOrNotDue extends BaseError {
+    static message = /aa21/
+    override name = "SmartAccountDoNotHaveEnoughFunds"
+    constructor({
+        cause,
+        docsPath
+    }: {
+        cause?: BaseError
+        docsPath?: string
+    }) {
+        super(
+            [
+                "The signature used in the user operation is not valid, because it is outside of the time range it specified.",
+                "",
+                "Possible reasons:",
+                "• This error occurs when the block.timestamp falls after the validUntil timestamp, or before the validAfter timestamp.",
+                "",
+                "Possible solutions:",
+                "• If you are looking to use time-based signatures, verify that the validAfter and validUntil fields are set correctly and that the user operation is sent within the specified range.",
+                "• If you are not looking to use time-based signatures, verify that the validAfter and validUntil fields are set to 0.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountRevertedOrOutOfGasDuringValidation extends BaseError {
+    static message = /aa23/
+    override name = "SmartAccountRevertedOrOutOfGasDuringValidation"
+    constructor({
+        cause,
+        sender,
+        docsPath
+    }: {
+        cause?: BaseError
+        sender: Address
+        docsPath?: string
+    }) {
+        super(
+            [
+                `The smart account ${sender} reverted or ran out of gas during the validation of the user operation.`,
+                "",
+                "Possible solutions:",
+                "• Verify that the verificationGasLimit is high enough to cover the validateUserOp function's gas costs.",
+                "• Make sure validateUserOp returns uint(1) for invalid signatures, and MUST NOT REVERT when the signature is invalid",
+                "• If you are not using a paymaster, verify that the sender address has enough native tokens to cover the required pre fund. Consider leveraging functions like getRequiredPrefund.",
+                "• Verify that the validateUserOp function is implemented with the correct logic, and that the user operation is supposed to be valid.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountInvalidSignature extends BaseError {
```

Comment:
SmartAccountInvalidSignature -> InvalidSignature?

Reply:
Similar, which signature is invalid